### ISSUE TITLE
Can't open/close by trigger node after change open prop

### ISSUE THREAD

**Issue by Sheile** [glennflanagan/react-collapsible#72]
### Description
If change status by two way which are trigger and open prop, unnecessary `openCollapsible` has been called.
after that, trigger does not work because `transitionend` will not be fired.

Already send pull request about it by other, it works for me.
Could you merge it?
https://github.com/glennflanagan/react-collapsible/pull/65

### Reproduce process
1. Click trigger text
2. Click toggle button
3. Can't open/close by trigger text

### Details
1. Click trigger text
 1.1. will open Collapsible(`height=24px`)
 1.2. `transitionend` will be occurred and `height=auto`
2. Click toggle button
 2.1. call unnecessary `openCollapsible` from `componentDidUpdate` by changing `open` prop
 2.2. call `continueOpenCollapsible` and `height=24px`
 2.3. `transitionend` WILL NOT BE FIRED because already opened
3. Can't open/close by trigger text
 3.1. call `closeCollapsible` and `height=24px`
 3.2. `window.setTimeout` WILL NOT BE CALLED because already `height` isn't `auto`

```
import React, { Component } from 'react'
import Collapsible from 'react-collapsible'

export class Sample extends Component {
  constructor() {
    super()
    this.state = { isOpen: false }
  }

  toggleCollapsible = () => {
    this.setState({ isOpen: !this.state.isOpen })
  }

  render() {
    return (
      <div>
        <Collapsible trigger="Trigger" open={this.state.isOpen}>
          <div>Lorem ipsum dolor sit amet, consectetur adipiscing elit</div>
        </Collapsible>
        <input type="button" onClick={this.toggleCollapsible} value="Toggle" />
      </div>
    )
  }
}

export default Sample
```

**Comment by Bezilon** [comment]
Exact same problem using:
react@16.3.1
react-dom@16.3.1

Not even changing the `open` prop in the React Dev Tool has any effect

**Comment by jpargana** [comment]
I have the same problem.

**Comment by karltaylor** [comment]
Managed to replicate but trying to figure out a solution #65 doesn't work for me as a solution. I have to click the toggle twice. Need to dig deeper into the problem.

**Comment by michael-ecb** [comment]
happen very fast with new react version:
react@16.4.1
react-dom@16.4.1

**Comment by michael-ecb** [comment]
@karltaylor it will be resolve ? 

thanks

**Comment by jcdiprose** [comment]
@michael-ecb switching to `onOpening` & `onClosing` solved the issue for me.

**Comment by karunalladeva** [comment]
I have the same problem.

**Comment by arye-dov-eidelman** [comment]
The open prop isn't a trigger button. It's a way to lift the open state up to a parent component (or anywhere really). ([react lift state up docs](https://reactjs.org/docs/lifting-state-up.html)).

By passing in the open prop you are telling the collapsible that you will keep track of the open state elsewhere and provide that state to the open prop. this can be useful if you need this state elsewhere (for example for an accordion or analytics).
in this case, you can use `onOpening`, `onClosing`, and any other logic to control the state.

**Comment by arye-dov-eidelman** [comment]
Make sure you are passing in a boolean. If you are passing in any other value you can add `!!` before it to coerce it into a boolean.

[Demo](https://jsfiddle.net/arye_eidelman/b8ev0k2u/87/)

**Comment by karunalladeva** [comment]
thanks

**Comment by karltaylor** [comment]
Is this issue now resolved? I'm wondering if it's fixed in v2.8.3. Will leave this open for a week or so but will close after to keep the repo clean.
